### PR TITLE
interactive: add "remove last pass" button (5/7)

### PR DESCRIPTION
**Pull request by dshaaban01**
add "remove last pass" button.

image

### CODE REVIEW

**Review by compor — APPROVED** [review]
LGTM

As a side note, your PR stacks are long and I find it difficult to keep in my "mental" stack what the ongoing change is.

This PR could work well as a standalone against `main`.

File: xdsl/interactive/app.tcss [hunk 1]
```diff
@@ -57,3 +63,16 @@ Screen {
 .condensed #uncondense_button {
     display: block;
 }
+
+# Button
```
Comment (superlopuh) [inline xdsl/interactive/app.tcss:67]: seems like duplicate from above

File: tests/interactive/test_app.py [hunk 2]
```diff
@@ -147,54 +147,54 @@ async def test_buttons():
...
+        expected_pass_pipeline = tuple(
+            (
+                *expected_pass_pipeline,
+                convert_func_to_riscv_func.ConvertFuncToRiscvFuncPass,
+            )
         )
-
-        # assert that pass selection affected Output Text Area
-        await pilot.pause()
+        assert app.pass_pipeline == expected_pass_pipeline
```
Comment (superlopuh) [inline tests/interactive/test_app.py:197]: nit:
```suggestion
        # press "Remove Last Pass" button
        current_pipeline = app.pass_pipeline
        await pilot.click("#remove_last_pass_button")
        await pilot.pause()
        assert app.pass_pipeline == current_pipeline[:-1]
```
feels a tiny bit neater to me

**Review by superlopuh — APPROVED** [review]
Am I right in understanding that the deleted test code is from the other PR removing redundant tests?

**Review by webmiche — APPROVED** [review]

**Pull request closed without merging**